lib/src/main/kotlin/tree/SearchTree.kt

```diff
@@ -1,157 +1,196 @@
+    protected var root: Node? = null
+    var size: Long = 0
+        private set
+    var recentlyKey: K? = null
+        private set
+
+    constructor(key: K, value: V) : this() {
+        set(key, value)
+    }
+
+    constructor(pairs: Array<Pair<K, V>>) : this() {
+        set(pairs)
+    }
+
+    protected abstract fun insertNode(node: Node)
+    protected abstract fun removeNode(node: Node)
+    protected abstract fun createNode(key: K, value: V): Node
+
+    /**
+     * Finding a node in a tree by key
+     */
+    protected fun searchNode(key: K): Node? {
+        TODO("Finding a node in a tree by key")
+    }
+
+    /**
+     * Stores the value for the given key. Return previous value.
+     */
+    fun set(key: K, value: V): V? {
+        TODO("Adding a node in a tree if there is no such node in the tree")
+    }
+
+    /**
+     * Stores the values for the given keys. Return previous values.
+     */
+    fun set(pairs: Array<Pair<K, V>>): Array<V?> {
+        TODO("Adding multiple node in a tree if there is no such node in the tree")
+    }
+
+    /**
+     * Stores the value for the given key if there is no pair with that key. Return previous value.
+     */
+    fun setIfEmpty(key: K, value: V): V? {
+        TODO("Adding a node if such a node already exists in the tree")
+    }
+
+    /**
+     * Stores the values for the given keys if there is no pair with that key. Return previous values.
+     */
+    fun setIfEmpty(pairs: Array<Pair<K, V>>): Array<V?> {
+        TODO("Adding several nodes if such a node already exists in the tree")
+    }
+
+    /**
+     * Remove the value for the given key. Return previous value.
+     */
+    fun remove(key: K): V? {
+        TODO("Adding multiple nodes if such nodes already exist in the tree")
+    }
+
+    /**
+     * Remove the values for the given keys. Return previous values.
+     */
+    fun remove(keys: Array<K>): Array<V?> {
+        TODO("Removing multiple catches by keys")
+    }
+
+    /**
+     * Return the value for the given key.
+     */
+    fun search(key: K): V? {
+        return searchNode(key)?.value
+    }
+
+    /**
+     * Returns a complete list of keys.
+     */
+    fun getKeys(): Array<K> {
+        TODO("Returns a complete list of keys")
+    }
+
+    /**
+     * Returns a complete list of values.
+     */
+    fun getValues(): Array<V> {
+        TODO("Returns a complete list of values")
+    }
+
+    /**
+     * Returns a complete list of pairs key value.
+     */
+    fun getEntities(): Array<Pair<K, V>> {
+        val result = emptyArray<Pair<K, V>>()
+        var index = 0
+
+        inOrderTraversal { result[index] = it; index++ }
+        return result
+    }
+
+    /**
+     * Returns pair with the minimum key.
+     */
+    fun getMin(): Pair<K?, V?> {
+        var node = root
+        while (node?.left != null) {
+            node = node.left
+        }
+        return Pair(node?.key, node?.value)
+    }

```

Comment:
Как по мне - новая строка перед return не помешает
```suggestion
    fun getMin(): Pair<K?, V?> {
        var node = root
        while (node?.left != null) {
            node = node.left
        }
        
        return Pair(node?.key, node?.value)
    }

```